app.js

```diff
@@ -394,6 +446,11 @@ function search() {
 }
 
 function onInput(e) {
+  var key = e.target.name || e.target.id;
```

Comment:
Do we rely on the target.id for any query string? AFAIK, your PR only uses the name attribute.